libs/ui-components/src/components/Device/DevicesPage/DeviceToolbarFilters.tsx

```diff
@@ -41,15 +41,26 @@ type LabelFleetSelectorProps = {
 };
 
 const LabelFleetResults = ({
+  filterText,
   allLabels,
   fleetNames,
   isUpdating,
 }: {
+  filterText: string;
   isUpdating: boolean;
   fleetNames: string[];
   allLabels: FlightCtlLabel[];
 }) => {
   const { t } = useTranslation();
+  const searchHighlighter = React.useCallback(
+    (text: string) => {
+      return text
+        .split(new RegExp(`(${filterText})`, 'g'))
+        .filter(Boolean)
+        .map((part, index) => (part.trim() === filterText ? <strong key={`part-${index}`}>{part}</strong> : part));
+    },
+    [filterText],
+  );
```

Comment:
Im not sure what the useCallback gives us here ? If we are after some performance optimization, IMO it would be more beneficial to not create a new regex in loop.

```suggestion
  const regexp = React.useMemo(() => new RegExp(`(${filterText})`, 'g'), [filterText]);
  const searchHighlighter = (text: string) => {
    return text
      .split(regexp)
      .filter(Boolean)
      .map((part, index) => (part.trim() === filterText ? <strong key={`part-${index}`}>{part}</strong> : part));
  };
```

Reply:
Yes, that was the idea. I agree with your suggestion, thanks